syft/frameworks/torch/tensors/interpreters/precision.py

```diff
@@ -9,7 +9,7 @@ def __init__(
         self,
         owner=None,
         id=None,
-        field: int = (2 ** 31) - 1,
+        field: int = (2 ** 62) - 1,
```

Comment:
This field is not prime

Reply:
In mathematics, a finite field or Galois field (so-named in honor of Évariste Galois) is a field that contains a finite number of elements. As with any field, a finite field is a set on which the operations of multiplication, addition, subtraction and division are defined and satisfy certain basic rules. The most common examples of finite fields are given by the integers mod p when p is a prime number.

Is from Wikipedia. From mortens blog the field needs to be a finite field

Reply:
The article of SecureNN claims that additions and multiplication should be done in a field of 2**32 or 2**64 as it allows to use the native modulo implementation fo the int and long types.
The field for the comparison in return (p=67 in the implementation & paper) should be prime

Reply:
Sounds good.

Reply:
I agree there is a lot of subtleties here which I don't address in terms of fields mainly because I keep allowing negative values - because long can be negative.
There is more work to do there but most of the protocol is now written :)